source/Halibut.Tests/Support/TestAttributes/CustomLevelOfParallelismAttribute.cs

```diff
@@ -28,23 +29,35 @@ public static int LevelOfParallelism()
                 return LevelOfParallelismInTeamCity();
             }
             
-            return LevelOfParallelismFromEnvVar()??NUnitTestAssemblyRunner.DefaultLevelOfParallelism * 2;
+            return LevelOfParallelismFromEnvVar() ?? NUnitTestAssemblyRunner.DefaultLevelOfParallelism * 2;
         }
 
         static int LevelOfParallelismInTeamCity()
         {
             if (NUnitTestAssemblyRunner.DefaultLevelOfParallelism <= 8)
             {
                 // Its unlikely a host with 8 cores is running multiple tests so assume we have access to them all.
+
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && NUnitTestAssemblyRunner.DefaultLevelOfParallelism > 6)
+                {
+                    return NUnitTestAssemblyRunner.DefaultLevelOfParallelism - 2;
```

Comment:
Can this ever become 0? 

`DefaultLevelOfParallelism` is defined as `Math.Max(Environment.ProcessorCount, 2)`, so it'll be 2 at a minimum. Therefore this could become 0.

I wonder if we also want to maintain the "2 as a minimum" here as well?

Reply:
Min it can be is 5

`RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && NUnitTestAssemblyRunner.DefaultLevelOfParallelism > 6`

Reply:
🤦 Yes, `NUnitTestAssemblyRunner.DefaultLevelOfParallelism > 6` line 41...